### PR TITLE
*: add executor runtime info for `explain for connection` statement (#19183)

### LINKED ISSUE #18233: Add executor runtime info for `explain for connection` statement
This is a subtask of https://github.com/pingcap/tidb/issues/17692

### background

Currently, you can use `explain for connection` statement to check the plan information of the running sql,  such as below:

```sql
 explain for connection 168;
+---------------------------+------------+-----------+------------------------+---------------------------------------------------------+
| id                        | estRows    | task      | access object          | operator info                                           |
+---------------------------+------------+-----------+------------------------+---------------------------------------------------------+
| HashAgg_11                | 1.00       | root      |                        | funcs:count(1)->Column#7                                |
| └─HashJoin_14             | 9659664.00 | root      |                        | CARTESIAN inner join, other cond:ne(test.t.b, test.t.b) |
|   ├─IndexReader_19(Build) | 3108.00    | root      |                        | index:IndexFullScan_18                                  |
|   │ └─IndexFullScan_18    | 3108.00    | cop[tikv] | table:t2, index:idx(b) | keep order:false                                        |
|   └─IndexReader_17(Probe) | 3108.00    | root      |                        | index:IndexFullScan_16                                  |
|     └─IndexFullScan_16    | 3108.00    | cop[tikv] | table:t1, index:idx(b) | keep order:false                                        |
+---------------------------+------------+-----------+------------------------+---------------------------------------------------------+
```

### Feature Request

Add executor runtime info for `explain for connection` statement, such as below:

```sql
+---------------------------+------------+---------+-----------+------------------------+---------------------------------------------------------------------------------+---------------------------------------------------------+------------------+---------+
| id                        | estRows    | actRows | task      | access object          | execution info                                                                  | operator info                                           | memory           | disk    |
+---------------------------+------------+---------+-----------+------------------------+---------------------------------------------------------------------------------+---------------------------------------------------------+------------------+---------+
| HashAgg_11                | 1.00       | 1       | root      |                        | time:2.75491619s, loops:2, PartialConcurrency:4, FinalConcurrency:4             | funcs:count(1)->Column#7                                | 0 Bytes          | N/A     |
| └─HashJoin_14             | 9659664.00 | 9656556 | root      |                        | time:2.39850105s, loops:9432, Concurrency:5, probe collision:0, build:434.295µs | CARTESIAN inner join, other cond:ne(test.t.b, test.t.b) | 32.9375 KB       | 0 Bytes |
|   ├─IndexReader_19(Build) | 3108.00    | 3108    | root      |                        | time:690.901112ms, loops:5, rpc num: 1, rpc time:1.134724315s, proc keys:6190   | index:IndexFullScan_18                                  | 24.5615234375 KB | N/A     |
|   │ └─IndexFullScan_18    | 3108.00    | 3108    | cop[tikv] | table:t2, index:idx(b) | time:117ms, loops:8                                                             | keep order:false                                        | N/A              | N/A     |
|   └─IndexReader_17(Probe) | 3108.00    | 3108    | root      |                        | time:732.663444ms, loops:5, rpc num: 1, rpc time:1.134706548s, proc keys:6190   | index:IndexFullScan_16                                  | 24.5634765625 KB | N/A     |
|     └─IndexFullScan_16    | 3108.00    | 3108    | cop[tikv] | table:t1, index:idx(b) | time:117ms, loops:8                                                             | keep order:false                                        | N/A              | N/A     |
+---------------------------+------------+---------+-----------+------------------------+---------------------------------------------------------------------------------+---------------------------------------------------------+------------------+---------+
```

Actually the upper result it the result of `explain analyze xxx`.

Currently, TiDB doesn't collect the executor runtime information by default, but you can enable it by set the variable:

```sql
set tidb_enable_collect_execution_info=1;
``` 

Then TiDB will record the executor runtime information in the `StatementContext.RuntimeStatsColl` of the session. So you can add this runtime information in `explain for connection` statement.

**Describe alternatives you've considered:**

We can refer to the implementation of `ExplainExec.generateExplainInfo`, to avoid duplicate code.

**Teachability, Documentation, Adoption, Migration Strategy:**

* [explain analyze](https://pingcap.com/docs-cn/dev/sql-statements/sql-statement-explain-analyze/)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #19183 to release-4.0

---

### What problem does this PR solve?

Issue Number: close #18233  

Problem Summary:
add this runtime information in explain for connection statement.

Example:

**When  set @@tidb_enable_collect_execution_info=1;**

```sql
test> show processlist;
+-----+------+-----------+------+---------+------+------------+-----------------------------------------------------------+
| Id  | User | Host      | db   | Command | Time | State      | Info                                                      |
+-----+------+-----------+------+---------+------+------------+-----------------------------------------------------------+
| 775 | root | 127.0.0.1 | test | Query   | 65   | autocommit | select count(*) from t1 tt1 join t1 tt2 where tt1.a>tt2.a |
| 1   | root | 127.0.0.1 | test | Query   | 0    | autocommit | show processlist                                          |
+-----+------+-----------+------+---------+------+------------+-----------------------------------------------------------+
test> explain for connection 775;
+---------------------------+-------------+---------+-----------+-----------------------+----------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------+--------+------+
| id                        | estRows     | actRows | task      | access object         | execution info                                                                                                       | operator info                                             | memory | disk |
+---------------------------+-------------+---------+-----------+-----------------------+----------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------+--------+------+
| HashAgg_11                | 1.00        | 0       | root      |                       | time:0s, loops:0                                                                                                     | funcs:count(1)->Column#7                                  | N/A    | N/A  |
| └─HashJoin_14             | 99800100.00 | 1083392 | root      |                       | time:11.67287184s, loops:1058                                                                                        | CARTESIAN inner join, other cond:gt(test.t1.a, test.t1.a) | N/A    | N/A  |
|   ├─IndexReader_19(Build) | 9990.00     | 482836  | root      |                       | time:138.962375ms, loops:473, cop_task: {num: 1, max:125.755703ms, proc_keys: 0, rpc_num: 1, rpc_time: 125.729723ms} | index:IndexFullScan_18                                    | N/A    | N/A  |
|   │ └─IndexFullScan_18    | 9990.00     | 0       | cop[tikv] | table:tt2, index:a(a) | time:0ns, loops:0                                                                                                    | keep order:false, stats:pseudo                            | N/A    | N/A  |
|   └─IndexReader_17(Probe) | 9990.00     | 5120    | root      |                       | time:126.260071ms, loops:5, cop_task: {num: 1, max:125.596953ms, proc_keys: 0, rpc_num: 1, rpc_time: 125.582473ms}   | index:IndexFullScan_16                                    | N/A    | N/A  |
|     └─IndexFullScan_16    | 9990.00     | 0       | cop[tikv] | table:tt1, index:a(a) | time:0ns, loops:0                                                                                                    | keep order:false, stats:pseudo                            | N/A    | N/A  |
+---------------------------+-------------+---------+-----------+-----------------------+----------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------+--------+------+

# after a while
test> explain for connection 775;
+---------------------------+-------------+-----------+-----------+-----------------------+----------------------------------------------------------------------------------+-----------------------------------------------------------+--------+------+
| id                        | estRows     | actRows   | task      | access object         | execution info                                                                   | operator info                                             | memory | disk |
+---------------------------+-------------+-----------+-----------+-----------------------+----------------------------------------------------------------------------------+-----------------------------------------------------------+--------+------+
| HashAgg_11                | 1.00        | 0         | root      |                       | time:0s, loops:0                                                                 | funcs:count(1)->Column#7                                  | N/A    | N/A  |
| └─HashJoin_14             | 99800100.00 | 145286144 | root      |                       | time:6m3.062011395s, loops:141881                                                | CARTESIAN inner join, other cond:gt(test.t1.a, test.t1.a) | N/A    | N/A  |
|   ├─IndexReader_19(Build) | 9990.00     | 482836    | root      |                       | time:138.962375ms, loops:473, cop_task: {num: 1, max:125.755703ms, proc_keys: 0} | index:IndexFullScan_18                                    | N/A    | N/A  |
|   │ └─IndexFullScan_18    | 9990.00     | 0         | cop[tikv] | table:tt2, index:a(a) | time:0ns, loops:0                                                                | keep order:false, stats:pseudo                            | N/A    | N/A  |
|   └─IndexReader_17(Probe) | 9990.00     | 19456     | root      |                       | time:127.068821ms, loops:19, cop_task: {num: 1, max:125.596953ms, proc_keys: 0}  | index:IndexFullScan_16                                    | N/A    | N/A  |
|     └─IndexFullScan_16    | 9990.00     | 0         | cop[tikv] | table:tt1, index:a(a) | time:0ns, loops:0                                                                | keep order:false, stats:pseudo                            | N/A    | N/A  |
+---------------------------+-------------+-----------+-----------+-----------------------+----------------------------------------------------------------------------------+-----------------------------------------------------------+--------+------+

```

**When  set @@tidb_enable_collect_execution_info=0;**, Since TiDB doesn't collect runtime information, so `explan for connection` will also not display the `executor info`.

```sql
explain for connection 778;
+---------------------------+-----------------+-----------+-----------------------+-----------------------------------------------------------+
| id                        | estRows         | task      | access object         | operator info                                             |
+---------------------------+-----------------+-----------+-----------------------+-----------------------------------------------------------+
| HashAgg_11                | 1.00            | root      |                       | funcs:count(1)->Column#7                                  |
| └─HashJoin_14             | 233130602896.00 | root      |                       | CARTESIAN inner join, other cond:gt(test.t1.a, test.t1.a) |
|   ├─IndexReader_19(Build) | 482836.00       | root      |                       | index:IndexFullScan_18                                    |
|   │ └─IndexFullScan_18    | 482836.00       | cop[tikv] | table:tt2, index:a(a) | keep order:false                                          |
|   └─IndexReader_17(Probe) | 482836.00       | root      |                       | index:IndexFullScan_16                                    |
|     └─IndexFullScan_16    | 482836.00       | cop[tikv] | table:tt1, index:a(a) | keep order:false                                          |
+---------------------------+-----------------+-----------+-----------------------+-----------------------------------------------------------+
```

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 
- Add executor information in `explain for connection` result.

### CODE REVIEW

File: executor/prepared_test.go [hunk 1]
```diff
@@ -164,8 +165,12 @@ func (s *testSuite12) TestPreparedStmtWithHint(c *C) {
 	c.Check(sm.killed, Equals, true)
 }
 
+<<<<<<< HEAD
 func (s *testSuite9) TestPlanCacheOnPointGet(c *C) {
 	defer testleak.AfterTest(c)()
+=======
```
Comment (qw4990) [inline executor/prepared_test.go:171]: Could you please resolve these conflicts? @hesper-guo

**Review by SunRunAway — APPROVED** [review]
LGTM

**Comment by SunRunAway** [comment]
@qw4990 PTAL

**Comment by SunRunAway** [comment]
@crazycs520 documentation is needed.

**Pull request merged**